### PR TITLE
Generate CarePlan should generate disease followup schedule

### PR DESCRIPTION
**Pull request by DebbieArita**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #[issue number] 

**Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/engine/src/main/java/org/smartregister/fhircore/engine/task/FhirCarePlanGenerator.kt [hunk 1]
```diff
@@ -103,6 +103,8 @@ constructor(
               "PlanDefinition.expression.language=${it.expression.language} not supported"
             )
 
+          fhirPathEngine.evaluate(null, input, null, subject, it.expression.expression)
+
```
Comment (dubdabasoduba) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/task/FhirCarePlanGenerator.kt:107]: @DebbieArita what are we adding this?

**Review by dubdabasoduba — CHANGES_REQUESTED** [review]

**Comment by ellykits** [comment]
@DebbieArita Kindly update this PR so we can have it merged.

File: android/engine/src/test/resources/plans/disease-followup/patient.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+      "family": "Hadi",
+      "given": [
+        "Bareera"
+      ]
+    }
+  ],
+  "gender": "female",
+  "birthDate": "2022-01-14",
+  "generalPractitioner":[{"reference":"Practitioner/TEST_PRACTITIONER"}]
+}
```
Comment (dubdabasoduba) [inline android/engine/src/test/resources/plans/disease-followup/patient.json:16]: Do we have a generic sampel patient that this test can use? That way we don't create a new patient resource for every test we write

File: android/quest/src/test/java/org/smartregister/fhircore/quest/StructureMapUtilitiesTest.kt [hunk 3]
```diff
@@ -79,6 +79,40 @@ class StructureMapUtilitiesTest : RobolectricTest() {
...
+    val pcm = FilesystemPackageCacheManager(true, ToolsVersion.TOOLS_VERSION)
+    // Package name manually checked from
+    // https://simplifier.net/packages?Text=hl7.fhir.core&fhirVersion=All+FHIR+Versions
+    val contextR4 = SimpleWorkerContext.fromPackage(pcm.loadPackage("hl7.fhir.r4.core", "4.0.1"))
+    contextR4.setExpansionProfile(Parameters())
+    contextR4.isCanRunWithoutTerminology = true
+
+    val transformSupportServices = TransformSupportServices(contextR4)
+    val scu = org.hl7.fhir.r4.utils.StructureMapUtilities(contextR4, transformSupportServices)
+    val map = scu.parse(immunizationStructureMap, "eCBIS Family Registration")
```
Comment (dubdabasoduba) [inline android/quest/src/test/java/org/smartregister/fhircore/quest/StructureMapUtilitiesTest.kt:97]: This name doesn't seem right

**Review by dubdabasoduba — CHANGES_REQUESTED** [review]

**Comment by dubdabasoduba** [comment]
@DebbieArita please add an appropriate title for this PR

**Review by dubdabasoduba — APPROVED** [review]

**Pull request merged**